addressbook/interface/exchangeAutoCompleteResult/mivExchangeAutoCompleteResult.js

```diff
@@ -116,8 +112,9 @@ mivExchangeAutoCompleteResult.prototype = {
      */
     //readonly attribute long defaultIndex;
     get defaultIndex() {
-        //dump("defaultIndex\n");
-        if (this._cards.length == 0) {
+        var defaultId = this._idcards.length;
```

Comment:
I don't understand this. The method name implies an index is returned but it is currently returning array length, therefore if you access with `arr[defaultIndex]` based on what the method name implies, you will get an out of bounds error.

I believe it would make more sense to return `this._idcards.length - 1` in which case it matches the behavior implied by the method name and you can also get rid of the subsequent `if` condition.

Reply:
If I understand correctly the [original interface](https://dxr.mozilla.org/comm-central/source/toolkit/components/autocomplete/nsIAutoCompleteResult.idl), the implementation of `nsIAutoCompleteResult.idl` should be simply an indexed array. Indeed all defined methods there just work with an index.

I've checked too and that's Exchange Calendar which defines the function `addResult` which doesn't work with index (that's used by `mivExchangeAutoCompleteSearch.js`).

So, here, we have to return an index that is inside the range of the current array.

I've just translated the old behavior with the new array:

- by default, we return always the first result of the list, so it's index is `0`
- however, if our array is empty, there's no index `0` and so we just return `-1`

We never return the array `length`, that's just used to check if the array is empty or not.

Reply:
I see. Then I would simply do `return this._idcards.length ? 0 : -1;` and get rid of the rest. Then it is short and obvious you don't actually care about the length.